models/AmazonS3.cfc

```diff
@@ -800,7 +800,12 @@ component accessors="true" singleton {
 			arguments.contentType = getFileMimeType( arguments.filepath );
 		}
 
-		var byteCount = getFileInfo( arguments.filepath ).size;
+		var byteCount = 0;
+		if( structKeyExists( variables, "asyncManager" )  ) {
```

Comment:
Let's allow multi-part uploads when AsyncManager is null ( e.g. outside Coldbox/CommandBox context ).  So, in the try/catch block for the multipart, we can do:

```
var processPart = function( part ){
	var channel = part.channel.position( part.offset );
	var buffer  = createObject( "java", "java.nio.ByteBuffer" ).allocate( part.limit );
	channel.read( buffer );

	return {
		"partNumber" : part.partNumber,
		"size"       : part.limit,
		"channel"    : part.channel,
		"response"   : s3Request(
			method     = "PUT",
			resource   = bucketName & "/" & uri,
			body       = buffer.array(),
			timeout    = part.timeout,
			parameters = {
				"uploadId"   : part.uploadId,
				"partNumber" : part.partNumber
			},
			headers = { "content-type" : "binary/octet-stream" }
		)
	};
};

if( !isNull( variables.asyncManagers ) ){
	variables.asyncManager.allApply( parts, processPart );
} else {
	parts = parts.map( processPart );
}
```